runtime/lua/vim/lsp.lua

```diff
@@ -1529,6 +1529,54 @@ function lsp.omnifunc(findstart, base)
   return -2
 end
 
+--- Provides an interface between the built-in client and a `formatexpr` function.
+---
+--- Currently only supports a single client. This can be set via
+--- `setlocal formatexpr=v:lua.vim.lsp.formatexpr()` but will typically or in `on_attach`
+--- via `vim.api.nvim_buf_set_option(bufnr, 'formatexpr', 'v:lua.vim.lsp.formatexpr()')`.
+---
+--- Can additionally be wrapped with a function that passes an optional table for customization.
```

Comment:
As `formatexpr` accepts an expression, `opts` can be given without wrapping:

`setlocal formatexpr=v:lua.vim.lsp.formatexpr(#{timeout_ms:123})`